### DISCUSSION TITLE
Are there plans to update the dotenv dependency to latest?

### DISCUSSION THREAD

**Discussion by dlindahl** [parcel-bundler/parcel#9344] [General]
Are there plans to update the `dotenv` dependency to latest? AFAICT, [`parcel/core` depends on 7.0.0](https://github.com/parcel-bundler/parcel/blob/493025912dec2bfc9cd2ded497b214f370d92015/packages/core/core/package.json#L46), but the [latest release of `dotenv` is 16.3.1](https://www.npmjs.com/package/dotenv). 

I'd like to be able to use [`dotenv-vault`](https://www.npmjs.com/package/dotenv-vault), but I can't get it to work by adding it as a dependency of my project since the `process.env` transforms have already run by the time my app loads.

**Comment by mischnic** [comment]
We can definitely do that (I guess nobody has thought about this until now), though it's the age old question of whether this is a breaking change for Parcel as the envfile parsing of dotenv changes when we upgrade. (Looks like backtick, hashtag and comment parsing changed.)

**Reply by dlindahl** [reply]
Yeah it is a tough call. I imagine since `dotenv` deemed it a breaking change (at some point between 7 and 16) that it might be worth following their lead.

I don't know if this is feasible, but is there a way to hook into the build process that would allow my own `dotenv` dependency to be used over the one that `parcel` has defined? I've tried using [`overrides` in the `package.json`](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#overrides), however, if I'm understanding the `parcel` source code correctly, [`parcel` never calls `dotenv.config()`](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#overrides) so that the vault code within `dotenv` is never run. In fact, it looks like its only using `dotenv` to parse a file that `parcel` has loaded on its own instead of letting `dotenv` do its thing. I imagine this is just how things needed to work back when v7 was the latest release, but it looks like `parcel` can do less by using more of the API that `dotenv` now provides.

Thanks for considering this update!
